modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -359,6 +359,11 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws I
                     "indexed document [{}/{}/{}] couldn't be found", indexedDocumentIndex, indexedDocumentType, indexedDocumentId
             );
         }
+        if(getResponse.isSourceEmpty()) {
+            throw new ElasticsearchException(
```

Comment:
Technically the _source doesn't exist, so a 404 could be returned. However the reason it is missing is because of source being disabled and then the 'percolate existing document' feature just can't work. For that reason I think 400 is a better response code, because that tells the client, that it is doing something that just doesn't work. So lets throw a `IllegalArgumentException ` here.

If a the document to be percolated is just missing, that document can be added, but enabling _source isn't possible on an existing index. So I think a 404 code is appropriate if the document to be percolated is missing.